Update Microsoft.AspNetCore.Watch.BrowserRefresh to net8.0

wtgodbe opened the pull request:
net8.0 is now the oldest supported TFM

wtgodbe committed:
Update Microsoft.AspNetCore.Watch.BrowserRefresh to net8.0

wtgodbe committed:
Update target framework to net8.0

ViktorHofer reviewed (approved):
@MichaelSimons this should allow the removal of the 6.0.2 targeting packs in SBRP.

wtgodbe commented:
Looks like this will require some code changes to react to the TFM update - @captainsafia could you take a look?

wtgodbe committed:
Update BrowserRefreshMiddleware.cs

wtgodbe commented:
@ViktorHofer 2 new prebuilts:

> /__w/1/vmr/repo-projects/Directory.Build.targets(924,5): warning : 2 new packages used not in baseline! See report at /__w/1/vmr/artifacts/prebuilt-report/baseline-comparison.xml for more information. Package IDs are: [/__w/1/vmr/repo-projects/dotnet.proj]
/__w/1/vmr/repo-projects/Directory.Build.targets(924,5): warning : Microsoft.AspNetCore.App.Ref.8.0.12 [/__w/1/vmr/repo-projects/dotnet.proj]
/__w/1/vmr/repo-projects/Directory.Build.targets(924,5): warning : Microsoft.NETCore.App.Ref.8.0.12 [/__w/1/vmr/repo-projects/dotnet.proj]

Do we need to bring back the SourceBuild bits? That version will always be the latest 8.0 unless we pin

wtgodbe commented:
Maybe we need to bring back 8.0.0 & pin to it: https://github.com/dotnet/source-build-reference-packages/pull/1178

wtgodbe committed:
Add .NET 8.0.0 targeting pack reference

wtgodbe commented:
https://github.com/dotnet/source-build-reference-packages/pull/1197

wtgodbe commented:
Next SBRP update should unblock

wtgodbe committed:
Add FrameworkReference for Microsoft.AspNetCore.App

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/source-build-reference-packages build 20250313.4

Microsoft.SourceBuild.Intermediate.source-build-reference-packages
 From Version 10.0.616001 -> To Version 10.0.616304

ViktorHofer commented:
> CSC(0,0): error CS0006: (NETCORE_ENGINEERING_TELEMETRY=Build) Metadata file '/__w/1/vmr/src/sdk/artifacts/sb/package-cache/microsoft.aspnetcore.app.ref/8.0.0/analyzers/dotnet/cs/Microsoft.AspNetCore.App.Analyzers.dll' could not be found

Looks like the aspnetcore targeting pack incorrectly lists analyzers? cc @dotnet/source-build

ViktorHofer committed:
Update dependency version in Version.Details.xml
